keras_cv/layers/preprocessing3d/random_pasting_objects.py

```diff
@@ -0,0 +1,242 @@
+            # Remove background point clouds that are in object_bounding_boxes.
+            existing_point_clouds_mask = ~points_in_paste_bounding_boxes[
+                frame_index, :
+            ] & tf.math.greater(point_clouds[frame_index, :, 3], 0.0)
+            existing_point_clouds = tf.boolean_mask(
+                point_clouds[frame_index], existing_point_clouds_mask, axis=0
+            )
+            paste_point_clouds = tf.boolean_mask(
+                object_point_clouds[frame_index],
+                tf.math.greater(object_point_clouds[frame_index, :, :, 3], 0.0),
+                axis=0,
+            )
+            point_clouds_list += [
+                tf.concat([paste_point_clouds, existing_point_clouds], axis=0)
+            ]
+
+            existing_bounding_boxes = tf.boolean_mask(
+                bounding_boxes[frame_index],
+                tf.math.greater(bounding_boxes[frame_index, :, 7], 0.0),
+            )
+            paste_bounding_boxes = tf.boolean_mask(
+                object_bounding_boxes[frame_index],
+                tf.math.greater(object_bounding_boxes[frame_index, :, 7], 0.0),
+                axis=0,
+            )
+            bounding_boxes_list += [
+                tf.concat([paste_bounding_boxes, existing_bounding_boxes], axis=0)
+            ]
+
+        point_clouds = tf.ragged.stack(point_clouds_list)
+        bounding_boxes = tf.ragged.stack(bounding_boxes_list)
+
+        return (
+            point_clouds.to_tensor(shape=original_point_clouds_shape),
+            bounding_boxes.to_tensor(shape=original_object_bounding_boxes),
+        )
+
+    def _augment(self, inputs):
+        result = inputs
+        point_clouds = inputs.get(POINT_CLOUDS, None)
+        bounding_boxes = inputs.get(BOUNDING_BOXES, None)
+        object_point_clouds = inputs.get(OBJECT_POINT_CLOUDS, None)
+        object_bounding_boxes = inputs.get(OBJECT_BOUNDING_BOXES, None)
+        transformation = self.get_random_transformation(
+            point_clouds=point_clouds,
+            bounding_boxes=bounding_boxes,
+            object_point_clouds=object_point_clouds,
+            object_bounding_boxes=object_bounding_boxes,
+        )
+        point_clouds, bounding_boxes = self.augment_point_clouds_bounding_boxes(
+            point_clouds,
+            bounding_boxes=bounding_boxes,
+            transformation=transformation,
+        )
+        result.update({POINT_CLOUDS: point_clouds, BOUNDING_BOXES: bounding_boxes})
+        return result
+
+    def call(self, inputs, training=True):
+        if training:
+            point_clouds = inputs[POINT_CLOUDS]
+            bounding_boxes = inputs[BOUNDING_BOXES]
+            if point_clouds.shape.rank == 3 and bounding_boxes.shape.rank == 3:
+                return self._augment(inputs)
+            elif point_clouds.shape.rank == 4 and bounding_boxes.shape.rank == 4:
+                batch = point_clouds.get_shape().as_list()[0]
```

Comment:
do we need to handle batch = None, i.e., when data.batch(drop_remainder=False)?

Reply:
For now, we only support batch != None. We could support `batch = None` once we fix the map_fn/vectorized_map bug.